### PR TITLE
NANs in xspectrum1d.rebin

### PR DESCRIPTION
**Pull request by profxj**
Handles NANs in flux array

Test too

### CODE REVIEW

File: linetools/spectra/tests/test_xspectrum_utils.py [hunk 1]
```diff
@@ -86,6 +86,10 @@ def test_rebin(spec):
     """
     imn = np.argmin(np.abs(newspec.wavelength-8055*u.AA))
     np.testing.assert_allclose(newspec.sig[imn].value, 0.0169634, rtol=1e-5)
+    # With NANs
+    spec.flux[100:110] = np.nan
```
Comment (ntejos) [inline linetools/spectra/tests/test_xspectrum_utils.py]: I do not think this works the way you are expecting (do not think you can reassign values this way).

**Comment by ntejos** [comment]
Looks good apart from the inline comment regarding the test.

**Comment by profxj** [comment]
Right you were.  Fixed.

**Comment by profxj** [comment]
Merging.

**Pull request merged**